lib/src/compatKtLint0Dot49Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot49Dot0Adapter.java

```diff
@@ -0,0 +1,210 @@
+import com.pinterest.ktlint.rule.engine.api.EditorConfigOverride;
+import com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine;
+import com.pinterest.ktlint.rule.engine.api.LintError;
+import com.pinterest.ktlint.rule.engine.core.api.Rule;
+import com.pinterest.ktlint.rule.engine.core.api.RuleProvider;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.CodeStyleEditorConfigPropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfigProperty;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EndOfLinePropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.IndentSizeEditorConfigPropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.IndentStyleEditorConfigPropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.InsertFinalNewLineEditorConfigPropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.MaxLineLengthEditorConfigPropertyKt;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.RuleExecution;
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.RuleExecutionEditorConfigPropertyKt;
+import com.pinterest.ktlint.ruleset.standard.StandardRuleSetProvider;
+
+import kotlin.Pair;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function2;
+
+public class KtLintCompat0Dot49Dot0Adapter implements KtLintCompatAdapter {
+
+	private static final Logger logger = LoggerFactory.getLogger(KtLintCompat0Dot49Dot0Adapter.class);
+
+	private static final List<EditorConfigProperty<?>> DEFAULT_EDITOR_CONFIG_PROPERTIES;
+
+	static {
+		List<EditorConfigProperty<?>> list = new ArrayList<>();
+		list.add(CodeStyleEditorConfigPropertyKt.getCODE_STYLE_PROPERTY());
+		list.add(EndOfLinePropertyKt.getEND_OF_LINE_PROPERTY());
+		list.add(IndentSizeEditorConfigPropertyKt.getINDENT_SIZE_PROPERTY());
+		list.add(IndentStyleEditorConfigPropertyKt.getINDENT_STYLE_PROPERTY());
+		list.add(InsertFinalNewLineEditorConfigPropertyKt.getINSERT_FINAL_NEWLINE_PROPERTY());
+		list.add(MaxLineLengthEditorConfigPropertyKt.getMAX_LINE_LENGTH_PROPERTY());
+		list.add(RuleExecutionEditorConfigPropertyKt.getEXPERIMENTAL_RULES_EXECUTION_PROPERTY());
+		DEFAULT_EDITOR_CONFIG_PROPERTIES = Collections.unmodifiableList(list);
+	}
+
+	private static final Method RULEID_METHOD;
+	private static final Method CREATE_RULESET_EXECUTION_METHOD;
+	private static final Method CREATE_RULE_EXECUTION_METHOD;
+
+	static {
+		try {
+			RULEID_METHOD = LintError.class.getMethod("getRuleId-6XN97os");
+			CREATE_RULESET_EXECUTION_METHOD = RuleExecutionEditorConfigPropertyKt.class.getMethod("createRuleSetExecutionEditorConfigProperty-fqiwTpU", String.class, RuleExecution.class);
+			CREATE_RULE_EXECUTION_METHOD = RuleExecutionEditorConfigPropertyKt.class.getMethod("createRuleExecutionEditorConfigProperty-U7AdEiY", String.class, RuleExecution.class);
+		} catch (NoSuchMethodException e) {
+			throw new RuntimeException(e);
+		}
```

Comment:
What is this?

Reply:
These are the mangled method names for `0.49.X`. They're unrepresentable in Java source because they contain a hyphen, and they're mangled in the first place because they deal with Kotlin `value` classes (and Valhalla isn't ready, so they implemented it like this first).
More info here: https://github.com/pinterest/ktlint/issues/2041